http/oidc/src/main/java/org/wildfly/security/http/oidc/TokenValidator.java

```diff
@@ -83,11 +85,18 @@ private TokenValidator(Builder builder) {
      * @throws OidcException if the ID token is invalid
      */
     public VerifiedTokens parseAndVerifyToken(final String idToken, final String accessToken) throws OidcException {
+        return parseAndVerifyToken(idToken, accessToken, null);
+    }
+
+    public VerifiedTokens parseAndVerifyToken(final String idToken, final String accessToken, final String stateCookieValue) throws OidcException {
         try {
             JwtContext idJwtContext = setVerificationKey(idToken, jwtConsumerBuilder);
             jwtConsumerBuilder.setExpectedAudience(clientConfiguration.getResourceName());
             jwtConsumerBuilder.registerValidator(new AzpValidator(clientConfiguration.getResourceName()));
             jwtConsumerBuilder.registerValidator(new AtHashValidator(accessToken, clientConfiguration.getTokenSignatureAlgorithm()));
+            if (stateCookieValue != null) {
```

Comment:
Since the state cookie value refers to something else we should be using a different variable name here.

Reply:
I provided a more generic name